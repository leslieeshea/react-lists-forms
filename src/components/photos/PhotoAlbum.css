.PhotoAlbum {
  padding: 1rem;
  margin: auto;

  h2 {
    font-family: monospace;
    font-size: 2rem;
    text-align: center;
    text-decoration: underline;
  }

  ul {
    display: flex;
    justify-content: space-evenly;
    flex-wrap: wrap;
    list-style-type: none;
  }

  li {
    height: 12rem;
    padding-bottom: 2rem;
  }

  img {
    width: auto;
    height: 100%;
  }
}
