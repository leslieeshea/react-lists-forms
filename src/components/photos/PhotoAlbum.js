import React from 'react';
import PropTypes from 'prop-types';
import Photos from './Photos';
import styles from './PhotoAlbum.css';

function PhotoAlbum({ title, photos }) {
  return (
    <section className={styles.PhotoAlbum}>
      <h2>{title}</h2>
      <Photos photos={photos} />
    </section>
  );
}

PhotoAlbum.propTypes = {
  title: PropTypes.string,
  photos: PropTypes.array.isRequired
};

PhotoAlbum.propTypes = {
  title: 'My Album'
};

export default PhotoAlbum;
