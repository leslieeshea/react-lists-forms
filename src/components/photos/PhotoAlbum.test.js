import React from 'react';
import { shallow } from 'enzyme';
import PhotoAlbum from './PhotoAlbum';

describe('PhotoAlbum component', () => {
  it('renders a PhotoAlbum', () => {
    const photos = [
      '/photo1',
      '/photo2',
      '/photo3'
    ];

    const wrapper = shallow(<PhotoAlbum title="My Photos" photos={photos} />);
    expect(wrapper).toMatchSnapshot();
  });
});
