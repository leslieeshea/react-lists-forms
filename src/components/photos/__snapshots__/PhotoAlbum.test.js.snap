// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoAlbum component renders a PhotoAlbum 1`] = `
<section
  className="PhotoAlbum"
>
  <h2>
    My Photos
  </h2>
  <Photos
    photos={
      Array [
        "/photo1",
        "/photo2",
        "/photo3",
      ]
    }
  />
</section>
`;
